test/base.test.js

```diff
@@ -455,3 +455,41 @@ test('should handle custom signer', async (t) => {
   t.equal(response.headers['set-cookie'].includes(DEFAULT_ENCRYPTED_SESSION_ID), false)
   t.match(response.headers['set-cookie'], /sessionId=[\w-]{32}.[\w-%]{43,135}; Path=\/; HttpOnly; Secure/)
 })
+
+test('should set session cookie if saveUninitialized is false and maxAge is on', async (t) => {
+  t.plan(2)
+  const DateNow = Date.now
+  const now = Date.now()
+  Date.now = () => now
+
+  const options = {
+    cookie: {
+      maxAge: 42
+    },
+    secret: DEFAULT_SECRET,
+    saveUninitialized: false
+  }
+  const plugin = fastifyPlugin(async (fastify, opts) => {
+    fastify.addHook('onRequest', (request, reply, done) => {
+      request.sessionStore.set(DEFAULT_SESSION_ID, {
+        // In this scenario, maxAge would have set expired in a previous request
```

Comment:
Does it worth testing it in a dedicated test case?

Reply:
We do here already:
https://github.com/fastify/session/blob/847826c6409931e107598fbf5096aac047603d3b/test/cookie.test.js#L98

But you're right, doing two requests in this test would make it less brittle.